Execution failed for task ':app:generateDebugServiceApolloSources'. > boolean

**Summary**
I got this error when generateApolloSources task.
```
Execution failed for task ':app:generateDebugServiceApolloSources'.
> boolean
```

**Version**
2.4.1

**Description**
I found that the error comes from a deprecated mutation in my schema.
```json
{
  "name": "stat_collect",
  "description": null,
  "args": [],
  "type": {
    "kind": "NON_NULL",
    "name": null,
    "ofType": {
      "kind": "LIST",
      "name": null,
      "ofType": {
        "kind": "NON_NULL",
        "name": null,
        "ofType": {
          "kind": "SCALAR",
          "name": "Boolean",
          "ofType": null
        }
      }
    }
  },
  "isDeprecated": true,
  "deprecationReason": null
}
```
And I have this graphql which uses the deprecated mutation:
```graphql
mutation Test {
    stat_collect
}
```
If the manually change is `isDeprecated` to `false`. The generateApolloSources task will success.

Here
> > Task :app:checkApolloVersions UP-TO-DATE
> > Task :app:generateDebugAndroidTestServiceApolloSources NO-SOURCE
> > Task :checkDebugAndroidTestServiceApolloDuplicates UP-TO-DATE
> > Task :app:generateDebugAndroidTestApolloSources UP-TO-DATE
> 
> > Task :app:generateDebugServiceApolloSources FAILED
> w: /Users/chalermpongsatayavibul/git/TestApollo/app/src/main/graphql/com/chalermpong/testapollo/Test.graphql:2:5: ApolloGraphQL: Use of deprecated field 'stat_collect'
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':app:generateDebugServiceApolloSources'.
> > boolean
> 
> * Try:
> Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Exception is:
> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:generateDebugServiceApolloSources'.
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:207)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$$Lambda$970/0000000000000000.accept(Unknown Source)
> 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:205)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
> 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
> 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
> 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
> 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
> 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
> 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
> 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
> 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$$Lambda$662/0000000000000000.execute(Unknown Source)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
> 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
> 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
> 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> Caused by: java.lang.AssertionError: boolean
> 	at com.squareup.javapoet.TypeName.box(TypeName.java:172)
> 	at com.apollographql.apollo.compiler.ClassNames.parameterizedListOf(ClassNames.kt:40)
> 	at com.apollographql.apollo.compiler.JavaTypeResolver.resolve(JavaTypeResolver.kt:23)
> 	at com.apollographql.apollo.compiler.JavaTypeResolver.resolve$default(JavaTypeResolver.kt:18)
> 	at com.apollographql.apollo.compiler.ir.Field.toTypeName(Field.kt:150)
> 	at com.apollographql.apollo.compiler.ir.Field.fieldSpec(Field.kt:83)
> 	at com.apollographql.apollo.compiler.SchemaTypeSpecBuilder.responseFieldSpecs(SchemaTypeSpecBuilder.kt:318)
> 	at com.apollographql.apollo.compiler.SchemaTypeSpecBuilder.classTypeSpec(SchemaTypeSpecBuilder.kt:105)
> 	at com.apollographql.apollo.compiler.SchemaTypeSpecBuilder.build(SchemaTypeSpecBuilder.kt:35)
> 	at com.apollographql.apollo.compiler.ir.Operation.toTypeSpec(Operation.kt:33)
> 	at com.apollographql.apollo.compiler.OperationTypeSpecBuilder.toTypeSpec(OperationTypeSpecBuilder.kt:52)
> 	at com.apollographql.apollo.compiler.ir.CodeGenerator$DefaultImpls.toTypeSpec$default(CodeGenerator.kt:6)
> 	at com.apollographql.apollo.compiler.GraphQLCompiler.writeJavaFiles(GraphQLCompiler.kt:252)
> 	at com.apollographql.apollo.compiler.GraphQLCompiler.write(GraphQLCompiler.kt:132)
> 	at com.apollographql.apollo.gradle.internal.ApolloGenerateSourcesTask.taskAction(ApolloGenerateSourcesTask.kt:185)
> 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
> 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
> 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
> 	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
> 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:726)
> 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:693)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:569)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
> 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:554)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:537)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:278)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:267)
> 	at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
> 	at org.gradle.internal.execution.steps.ExecuteStep$$Lambda$821/0000000000000000.get(Unknown Source)
> 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
> 	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
> 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
> 	at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
> 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
> 	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
> 	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
> 	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
> 	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
> 	at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
> 	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
> 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
> 	at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
> 	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
> 	at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
> 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
> 	at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
> 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
> 	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
> 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
> 	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
> 	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
> 	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
> 	at org.gradle.internal.execution.steps.SkipUpToDateStep$$Lambda$574/0000000000000000.apply(Unknown Source)
> 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
> 	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
> 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
> 	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
> 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
> 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
> 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
> 	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
> 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
> 	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
> 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
> 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
> 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep$$Lambda$462/0000000000000000.get(Unknown Source)
> 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
> 	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
> 	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
> 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
> 	at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
> 	at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:194)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
> 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
> 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
> 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
> 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
> 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
> 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
> 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
> 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$$Lambda$662/0000000000000000.execute(Unknown Source)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
> 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
> 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
> 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> 
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 703ms
> 3 actionable tasks: 1 executed, 2 up-to-date

--- a/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ClassNames.kt
+++ b/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ClassNames.kt
@@ -37,7 +37,13 @@ object ClassNames {
       ParameterizedTypeName.get(LIST, ClassName.get(type))
 
   fun parameterizedListOf(typeArgument: TypeName): TypeName =
-      ParameterizedTypeName.get(LIST, typeArgument.let { if (it.isPrimitive) it.box() else it.withoutAnnotations() })
+      ParameterizedTypeName.get(LIST, typeArgument.let {
+        if (it.isPrimitive) {
+          it.box()
+        } else {
+          it.withoutAnnotations()
+        }
+      })
 
   fun <K : Any, V : Any> parameterizedMapOf(keyTypeArgument: Class<K>, valueTypeArgument: Class<V>): TypeName =
       ParameterizedTypeName.get(MAP, ClassName.get(keyTypeArgument).withoutAnnotations(),

--- a/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/JavaTypeResolver.kt
+++ b/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/JavaTypeResolver.kt
@@ -14,13 +14,23 @@ class JavaTypeResolver(
     private val packageName: String,
     private val deprecated: Boolean = false
 ) {
-  fun resolve(typeName: String, isOptional: Boolean = !typeName.endsWith("!"),
-      nullableValueType: NullableValueType? = null): TypeName {
+  fun resolve(
+      typeName: String,
+      isOptional: Boolean = !typeName.endsWith("!"),
+      nullableValueType: NullableValueType? = null,
+      supportsAnnotations: Boolean = true
+  ): TypeName {
     val normalizedTypeName = typeName.removeSuffix("!")
     val isList = normalizedTypeName.startsWith('[') && normalizedTypeName.endsWith(']')
     val customScalarType = context.customTypeMap[normalizedTypeName]
     val javaType = when {
-      isList -> ClassNames.parameterizedListOf(resolve(normalizedTypeName.removeSurrounding("[", "]"), false))
+      isList -> ClassNames.parameterizedListOf(
+          resolve(
+              typeName = normalizedTypeName.removeSurrounding("[", "]"),
+              isOptional = false,
+              supportsAnnotations = false
+          )
+      )
       normalizedTypeName == ScalarType.STRING.name -> ClassNames.STRING
       normalizedTypeName == ScalarType.INT.name -> if (isOptional) TypeName.INT.box() else TypeName.INT
       normalizedTypeName == ScalarType.BOOLEAN.name -> if (isOptional) TypeName.BOOLEAN.box() else TypeName.BOOLEAN
@@ -29,21 +39,33 @@ class JavaTypeResolver(
       else -> ClassName.get(packageName, normalizedTypeName.capitalize())
     }
 
-    return if (javaType.isPrimitive) {
-      javaType.let { if (deprecated) it.annotated(Annotations.DEPRECATED) else it }
-    } else if (isOptional) {
-      when (nullableValueType ?: context.nullableValueType) {
-        NullableValueType.APOLLO_OPTIONAL -> parameterizedOptional(javaType)
-        NullableValueType.GUAVA_OPTIONAL -> parameterizedGuavaOptional(javaType)
-        NullableValueType.JAVA_OPTIONAL -> parameterizedJavaOptional(javaType)
-        NullableValueType.INPUT_TYPE -> parameterizedInputType(javaType)
-        else -> javaType.annotated(Annotations.NULLABLE)
-      }.let {
-        if (deprecated) it.annotated(Annotations.DEPRECATED) else it
+    return when {
+      !supportsAnnotations -> {
+        // Adding an annotation to a primitive type will crash in javapoet's TypeName.box()
+        // The underlying issue is that the annotation should be on the field name and not on the type name but it is a more
+        // complex issue. Don't add the annotation for now if the type is used in a context where annotations do not make sense.
+        // See https://github.com/apollographql/apollo-android/issues/2660
+        // See https://github.com/apollographql/apollo-android/issues/2246
+        javaType
       }
-    } else {
-      javaType.annotated(Annotations.NONNULL).let {
-        if (deprecated) it.annotated(Annotations.DEPRECATED) else it
+      javaType.isPrimitive -> {
+        javaType.let { if (deprecated) it.annotated(Annotations.DEPRECATED) else it }
+      }
+      isOptional -> {
+        when (nullableValueType ?: context.nullableValueType) {
+          NullableValueType.APOLLO_OPTIONAL -> parameterizedOptional(javaType)
+          NullableValueType.GUAVA_OPTIONAL -> parameterizedGuavaOptional(javaType)
+          NullableValueType.JAVA_OPTIONAL -> parameterizedJavaOptional(javaType)
+          NullableValueType.INPUT_TYPE -> parameterizedInputType(javaType)
+          else -> javaType.annotated(Annotations.NULLABLE)
+        }.let {
+          if (deprecated) it.annotated(Annotations.DEPRECATED) else it
+        }
+      }
+      else -> {
+        javaType.annotated(Annotations.NONNULL).let {
+          if (deprecated) it.annotated(Annotations.DEPRECATED) else it
+        }
       }
     }
   }

--- a/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.java
+++ b/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.java
@@ -27,6 +27,8 @@ import com.example.mutation_create_review.type.CustomType;
 import com.example.mutation_create_review.type.Episode;
 import com.example.mutation_create_review.type.ReviewInput;
 import java.io.IOException;
+import java.lang.Boolean;
+import java.lang.Deprecated;
 import java.lang.Object;
 import java.lang.Override;
 import java.lang.String;
@@ -43,7 +45,7 @@ import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpisode.Data, Optional<CreateReviewForEpisode.Data>, CreateReviewForEpisode.Variables> {
-  public static final String OPERATION_ID = "c07e5abc4b4070cd773623194c07f546e609af467a1d34f7bf01c37272245296";
+  public static final String OPERATION_ID = "ddb5b88794340f1233d8b0d93bf51320b4e285a912f0914ddbaba8fc430e6db7";
 
   public static final String QUERY_DOCUMENT = QueryDocumentMinifier.minify(
     "mutation CreateReviewForEpisode($ep: Episode!, $review: ReviewInput!) {\n"
@@ -59,6 +61,7 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
         + "      name\n"
         + "    }\n"
         + "  }\n"
+        + "  stat_collect\n"
         + "}"
   );
 
@@ -236,10 +239,13 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
         .put("kind", "Variable")
         .put("variableName", "review")
         .build())
-      .build(), true, Collections.<ResponseField.Condition>emptyList())
+      .build(), true, Collections.<ResponseField.Condition>emptyList()),
+      ResponseField.forList("stat_collect", "stat_collect", null, false, Collections.<ResponseField.Condition>emptyList())
     };
 
     final Optional<CreateReview> createReview;
+
+    final @NotNull @Deprecated List<Boolean> stat_collect;
 
     private transient volatile String $toString;
 
@@ -247,12 +253,21 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
 
     private transient volatile boolean $hashCodeMemoized;
 
-    public Data(@Nullable CreateReview createReview) {
+    public Data(@Nullable CreateReview createReview,
+        @NotNull @Deprecated List<Boolean> stat_collect) {
       this.createReview = Optional.fromNullable(createReview);
+      this.stat_collect = Utils.checkNotNull(stat_collect, "stat_collect == null");
     }
 
     public Optional<CreateReview> createReview() {
       return this.createReview;
+    }
+
+    /**
+     * For testing https://github.com/apollographql/apollo-android/issues/2660
+     */
+    public @NotNull @Deprecated List<Boolean> stat_collect() {
+      return this.stat_collect;
     }
 
     @SuppressWarnings({"rawtypes", "unchecked"})
@@ -261,6 +276,14 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
         @Override
         public void marshal(ResponseWriter writer) {
           writer.writeObject($responseFields[0], createReview.isPresent() ? createReview.get().marshaller() : null);
+          writer.writeList($responseFields[1], stat_collect, new ResponseWriter.ListWriter() {
+            @Override
+            public void write(List items, ResponseWriter.ListItemWriter listItemWriter) {
+              for (Object item : items) {
+                listItemWriter.writeBoolean((Boolean) item);
+              }
+            }
+          });
         }
       };
     }
@@ -269,7 +292,8 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
     public String toString() {
       if ($toString == null) {
         $toString = "Data{"
-          + "createReview=" + createReview
+          + "createReview=" + createReview + ", "
+          + "stat_collect=" + stat_collect
           + "}";
       }
       return $toString;
@@ -282,7 +306,8 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
       }
       if (o instanceof Data) {
         Data that = (Data) o;
-        return this.createReview.equals(that.createReview);
+        return this.createReview.equals(that.createReview)
+         && this.stat_collect.equals(that.stat_collect);
       }
       return false;
     }
@@ -293,6 +318,8 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
         int h = 1;
         h *= 1000003;
         h ^= createReview.hashCode();
+        h *= 1000003;
+        h ^= stat_collect.hashCode();
         $hashCode = h;
         $hashCodeMemoized = true;
       }
@@ -310,7 +337,13 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
             return createReviewFieldMapper.map(reader);
           }
         });
-        return new Data(createReview);
+        final List<Boolean> stat_collect = reader.readList($responseFields[1], new ResponseReader.ListReader<Boolean>() {
+          @Override
+          public Boolean read(ResponseReader.ListItemReader listItemReader) {
+            return listItemReader.readBoolean();
+          }
+        });
+        return new Data(createReview, stat_collect);
       }
     }
   }

--- a/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.java
+++ b/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.java
@@ -265,6 +265,7 @@ public final class CreateReviewForEpisode implements Mutation<CreateReviewForEpi
 
     /**
      * For testing https://github.com/apollographql/apollo-android/issues/2660
+     * @deprecated Deprecated
      */
     public @NotNull @Deprecated List<Boolean> stat_collect() {
       return this.stat_collect;

--- a/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.kt
+++ b/apollo-compiler/src/test/graphql/com/example/mutation_create_review/CreateReviewForEpisode.kt
@@ -27,6 +27,7 @@ import java.util.Date
 import kotlin.Any
 import kotlin.Array
 import kotlin.Boolean
+import kotlin.Deprecated
 import kotlin.Int
 import kotlin.String
 import kotlin.Suppress
@@ -270,10 +271,19 @@ internal data class CreateReviewForEpisode(
    * Data from the response after executing this GraphQL operation
    */
   data class Data(
-    val createReview: CreateReview?
+    val createReview: CreateReview?,
+    /**
+     * For testing https://github.com/apollographql/apollo-android/issues/2660
+     */
+    @Deprecated(message = "Deprecated")
+    val stat_collect: List<Boolean>
   ) : Operation.Data {
     override fun marshaller(): ResponseFieldMarshaller = ResponseFieldMarshaller.invoke { writer ->
       writer.writeObject(RESPONSE_FIELDS[0], this@Data.createReview?.marshaller())
+      writer.writeList(RESPONSE_FIELDS[1], this@Data.stat_collect) { value, listItemWriter ->
+        value?.forEach { value ->
+          listItemWriter.writeBoolean(value)}
+      }
     }
 
     companion object {
@@ -284,15 +294,20 @@ internal data class CreateReviewForEpisode(
               "variableName" to "ep"),
             "review" to mapOf<String, Any>(
               "kind" to "Variable",
-              "variableName" to "review")), true, null)
+              "variableName" to "review")), true, null),
+          ResponseField.forList("stat_collect", "stat_collect", null, false, null)
           )
 
       operator fun invoke(reader: ResponseReader): Data = reader.run {
         val createReview = readObject<CreateReview>(RESPONSE_FIELDS[0]) { reader ->
           CreateReview(reader)
         }
+        val stat_collect = readList<Boolean>(RESPONSE_FIELDS[1]) { reader ->
+          reader.readBoolean()
+        }!!.map { it!! }
         Data(
-          createReview = createReview
+          createReview = createReview,
+          stat_collect = stat_collect
         )
       }
 
@@ -303,7 +318,7 @@ internal data class CreateReviewForEpisode(
 
   companion object {
     const val OPERATION_ID: String =
-        "c07e5abc4b4070cd773623194c07f546e609af467a1d34f7bf01c37272245296"
+        "ddb5b88794340f1233d8b0d93bf51320b4e285a912f0914ddbaba8fc430e6db7"
 
     val QUERY_DOCUMENT: String = QueryDocumentMinifier.minify(
           """
@@ -320,6 +335,7 @@ internal data class CreateReviewForEpisode(
           |      name
           |    }
           |  }
+          |  stat_collect
           |}
           """.trimMargin()
         )

--- a/apollo-compiler/src/test/graphql/com/example/mutation_create_review/TestOperation.graphql
+++ b/apollo-compiler/src/test/graphql/com/example/mutation_create_review/TestOperation.graphql
@@ -9,4 +9,5 @@ mutation CreateReviewForEpisode($ep: Episode!, $review: ReviewInput!) {
       name
     }
   }
+  stat_collect
 }

--- a/apollo-compiler/src/test/graphql/schema.sdl
+++ b/apollo-compiler/src/test/graphql/schema.sdl
@@ -263,6 +263,8 @@ enum LengthUnit {
 """The mutation type, represents all updates we can make to our data"""
 type Mutation {
   createReview(episode: Episode, review: ReviewInput!): Review
+  """For testing https://github.com/apollographql/apollo-android/issues/2660"""
+  stat_collect: [Boolean!]! @deprecated
 }
 
 """Information for paginating this connection"""

--- a/apollo-compiler/src/test/graphql/schema.sdl
+++ b/apollo-compiler/src/test/graphql/schema.sdl
@@ -264,7 +264,7 @@ enum LengthUnit {
 type Mutation {
   createReview(episode: Episode, review: ReviewInput!): Review
   """For testing https://github.com/apollographql/apollo-android/issues/2660"""
-  stat_collect: [Boolean!]! @deprecated
+  stat_collect: [Boolean!]! @deprecated(reason: "Deprecated")
 }
 
 """Information for paginating this connection"""

